src/V3Param.cpp

```diff
@@ -330,6 +330,9 @@ class ParamProcessor final {
             key += " ";
             key += paramValueKey(dtypep->subDTypep());
         } else if (const AstBasicDType* const dtypep = VN_CAST(nodep, BasicDType)) {
+            if (dtypep->isSigned()) {
+                key += " signed";
+            }
```

Comment:
A logical attempt at a fix, but this isn't the problem; all internal data types are always unsigned. This wouldn't fix other related problems like signed multiply. In V3Width visit_cmp_eq_gt the < (AstLt) should be converted to a signed flavor (AstLts).

Reply:
And for some reason it isn't, so if you'd like to look into why and update the patch that's appreciated.

Reply:
I suspect you will see it if use * in place of +.  Anyhow, the operator type needs to be correct for signed.

Reply:
I'm not sure what you mean by using * in place of + since I don't think I'm adding anything in my test, but I haven't been able to create a test that demonstrates an issue with the sign of *.

The thing with my test is that the output changes based on the order of instantiation of the cmp modules with the second module producing incorrect results, so I don't follow how there's only an issue with signed operations here.

Reply:
I think I misanalyzed my concern, this should be good to go.